### DISCUSSION TITLE
Integrating with a 3rd party editor.

### DISCUSSION THREAD

**Discussion by ltdrdata** [comfyanonymous/ComfyUI#300] [Ideas]
At present, in order to use **inpaint** and **controlnet**, it is necessary to edit the image and save it as a file, and then set the LoadImage node manually.

I believe we can offer a more convenient solution by utilizing OpenAPI and making a few adjustments to a third-party editor.

For example.
1. Add a 'launch openpose editor' button on the LoadImage node.
2. Launch the 3rd party tool and pass the updating node id as a parameter on click.
3. Implement the openapi for LoadImage updating.
4. Add the feature of receiving the node id and sending the updated image data from the 3rd party editor to ComfyUI through openapi.

By completing these steps, we can easily achieve the desired solution.

So. I am considering this project as candidate.

https://github.com/ZhUyU1997/open-pose-editor

**Comment by WASasquatch** [comment]
Also conveniently, we're already using canvas, with plenty of modules about for doing painting with various brushes, layering effects, etc. A node could simply act as a painting canvas input. Deadspace on bottom of nodes could easily accommodate a anchored toolbar.
